### ISSUE TITLE
Waypoint doesn't fire after switching url

### ISSUE THREAD

**Issue by SnowDiamond** [brigade/react-waypoint#171]
I'm trying to implement infinite scroll with fetching data from server. On initial page load **with fully browser reload** everything works as expected, so waypoint fires every time when I scroll to page bottom. But if I go to page with infinite scroll from any another page (using react-router) then waypoint fires only once at page load, and never again when scrolling to page bottom. Could you please help me to figure out what can be the problem?

My code is:

```
import React from 'react'
import UserItem from './UserItem'
import Waypoint from 'react-waypoint'

export default (props) => {
  return (
    <div>
      <div className='users-search-result clearfix'>
        {props.users.map(user => <UserItem key={user.id} user={user} />)}
      </div>
      <Waypoint onEnter={props.onPageScroll}>
        <div className='loader' />
      </Waypoint>
    </div>
  )
}
```

**Comment by trotzig** [comment]
Is this component remounted in between page transitions? If it's not, then the waypoint will not trigger on page load. You could potentially work around that with a `key` prop set on the waypoint, something unique for the page: 

```jsx
  <Waypoint key={this.props.foo} />
```

You can also avoid rendering the waypoint during load, which would cause it to unmount and remount again: 

```jsx
{!this.state.loading && <Waypoint onEnter={() => this.setState({ loading: true })} />
```

Let me know if this helps!

**Comment by SnowDiamond** [comment]
@trotzig  thank you for your help! Component is remounted between transitions, so componentDidMount fires every time during transitions. 
I also tried to specify `key`, like you suggested but it doesn't help. Any other thoughts? :)

**Comment by trotzig** [comment]
You could add the `debug` prop and look at what's going on in the console. It might be that it's always in view. 

```jsx
<Waypoint debug />
```

**Comment by SnowDiamond** [comment]
Maybe I explain my problem not clear. Problem is that waypoint **doesn't fires at all** when I scroll to page bottom, so nothing to debug. And this happens only if I go to the page from any another page.  But if I fully reload browser on the page it works correct, and fires each time when I scroll to the bottom.

**Comment by trotzig** [comment]
I re-read your initial comment now. That was clear from the start, I was just a little quick to come to conclusions. Sorry!

The `debug` thing  is still interesting though, it should give us a little more input. Try with a full page reload and with a react-router transition, then compare the different logs.

**Comment by SnowDiamond** [comment]
Debug log with full browser reload:
```
["scrollableAncestor", Window]
debugLog.js:9["waypoint top", 167.875]
debugLog.js:9["waypoint bottom", 167.875]
debugLog.js:9["scrollableAncestor height", 301]
debugLog.js:9["scrollableAncestor scrollTop", 0]
debugLog.js:9["currentPosition", "inside"]
debugLog.js:9["previousPosition", undefined]
debugLog.js:9["waypoint top", 167.875]
debugLog.js:9["waypoint bottom", 167.875]
debugLog.js:9["scrollableAncestor height", 301]
debugLog.js:9["scrollableAncestor scrollTop", 0]
debugLog.js:9["currentPosition", "inside"]
debugLog.js:9["previousPosition", "inside"]
debugLog.js:9["waypoint top", 79]
debugLog.js:9["waypoint bottom", 79]
debugLog.js:9["scrollableAncestor height", 301]
debugLog.js:9["scrollableAncestor scrollTop", 0]
debugLog.js:9["currentPosition", "inside"]
debugLog.js:9["previousPosition", "inside"]
debugLog.js:9["waypoint top", 71]
debugLog.js:9["waypoint bottom", 71]
debugLog.js:9["scrollableAncestor height", 301]
debugLog.js:9["scrollableAncestor scrollTop", 0]
debugLog.js:9["currentPosition", "inside"]
debugLog.js:9["previousPosition", "inside"]
debugLog.js:9 ["waypoint top", 1429]
debugLog.js:9 ["waypoint bottom", 1429]
debugLog.js:9 ["scrollableAncestor height", 301]
debugLog.js:9 ["scrollableAncestor scrollTop", 0]
debugLog.js:9 ["currentPosition", "below"]
debugLog.js:9 ["previousPosition", "inside"]
```

Debug log with react-router transition:
```
["scrollableAncestor", body]
debugLog.js:9 ["waypoint top", 71]
debugLog.js:9 ["waypoint bottom", 71]
debugLog.js:9 ["scrollableAncestor height", 633]
debugLog.js:9 ["scrollableAncestor scrollTop", 0]
debugLog.js:9 ["currentPosition", "inside"]
debugLog.js:9 ["previousPosition", undefined]
debugLog.js:9 ["waypoint top", 71]
debugLog.js:9 ["waypoint bottom", 71]
debugLog.js:9 ["scrollableAncestor height", 633]
debugLog.js:9 ["scrollableAncestor scrollTop", 0]
debugLog.js:9 ["currentPosition", "inside"]
debugLog.js:9 ["previousPosition", "inside"]
debugLog.js:9 ["waypoint top", 71]
debugLog.js:9 ["waypoint bottom", 71]
debugLog.js:9 ["scrollableAncestor height", 633]
debugLog.js:9 ["scrollableAncestor scrollTop", 0]
debugLog.js:9 ["currentPosition", "inside"]
debugLog.js:9 ["previousPosition", "inside"]
debugLog.js:9 ["waypoint top", 71]
debugLog.js:9 ["waypoint bottom", 71]
debugLog.js:9 ["scrollableAncestor height", 633]
debugLog.js:9 ["scrollableAncestor scrollTop", 0]
debugLog.js:9 ["currentPosition", "inside"]
debugLog.js:9 ["previousPosition", "inside"]
debugLog.js:9 ["waypoint top", 1429]
debugLog.js:9 ["waypoint bottom", 1429]
debugLog.js:9 ["scrollableAncestor height", 1549]
debugLog.js:9 ["scrollableAncestor scrollTop", 0]
debugLog.js:9 ["currentPosition", "inside"]
debugLog.js:9 ["previousPosition", "inside"]
```

So, I can notice that `scrollableAncestor` different, and also indicators has different values.

**Comment by SnowDiamond** [comment]
I added `scrollableAncestor={window}` and now it work correct! @trotzig  thank you a lot! :)

**Comment by trotzig** [comment]
Very interesting, glad you got it sorted! I wonder why the scrollable ancestor changed? This might be a bug in the waypoint code.

**Comment by JustinMoser** [comment]
Hi - Sorry to comment on this after such a long time, but I am having the exact same problem with Waypoint. Im using the latest version, however when changing pages via react router, the onEnter function doesnt get called when reaching the waypoint

Should I open another ticket?

Thanks!
